bundle/jsky.app.ot/src/main/scala/jsky/app/ot/ags/BagsManager.scala

```diff
@@ -166,50 +189,61 @@ object BagsManager {
     }
   }
 
-  // If a BAGS task failed to be processed properly, we should requeue it.
-  private def taskComplete(task: BagsTask, success: Boolean) = {
+  // If a BAGS task succeeded, process it accordingly. This can comprise one of two things:
+  // 1. If the task is marked as done, remove it from the map.
+  // 2. If the task is marked as not done, requeue it.
+  private def taskSuccess(task: BagsTask): Unit = {
     val obsKey = task.obsKey
-    if (success) {
-      // Check to see if the task is marked as done, and if so, remove.
-      val doneWithTask = taskMap.synchronized {
-        taskMap.get(obsKey).fold(true){t =>
-          val done = t.isDone
-          if (done) taskMap -= obsKey
-          done
-        }
-      }
-
-      // If a task is not complete and the program is still active, we add it back.
-      if (!doneWithTask) {
-        val isActiveProgram = activePrograms.synchronized(activePrograms.contains(task.observation.getProgramKey))
-        if (isActiveProgram) {
-          taskQueue.synchronized {
-            if (!taskQueue.contains(obsKey))
-              taskQueue.add(obsKey)
-          }
-        }
+    // Check to see if the task is marked as done, and if so, remove.
+    val doneWithTask = taskMap.synchronized {
+      taskMap.get(obsKey).fold(true) { t =>
+        val done = t.isDone
+        if (done) taskMap -= obsKey
+        done
       }
     }
-    else {
+
+    // If a task is not complete and the program is still active, we add it back.
+    if (!doneWithTask) {
       val isActiveProgram = activePrograms.synchronized(activePrograms.contains(task.observation.getProgramKey))
       if (isActiveProgram) {
-        Thread.sleep(RequeueDelay)
-
-        // We only need to do something if the task is not already in the queue.
         taskQueue.synchronized {
-          if (!taskQueue.contains(obsKey)) {
-            val newTask = task match {
-              case BagsTask(obs, Some(ctx), None) => BagsTask(obs, None, Some(ctx))
-              case t => t
-            }
-            taskMap.synchronized(taskMap += ((obsKey, newTask)))
+          if (!taskQueue.contains(obsKey))
             taskQueue.add(obsKey)
-          }
         }
       }
     }
   }
 
+  // Called when a BAGS task failed to be processed properly, in which case, we must requeue it.
```

Comment:
There isn't. It would be quite simple to add, but not ideal in any sense, as if network connectivity were to, say, come back, then the BAGS task would be dead.

Maybe a better thing to do would be to include a delay directly in BAGS tasks that fail and wait on that instead of what I am doing now. Indeed, I think that would be preferable and prevent them from being queued up again right away.